hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/InvocationNestedLocalTest.java

```diff
@@ -8,115 +8,120 @@
 import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Rule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
 
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
 public class InvocationNestedLocalTest extends InvocationNestedTest {
 
-    private final String response = "someresponse";
+    private static final String RESPONSE = "someresponse";
+
+    @Rule
+    public ExpectedException expected = ExpectedException.none();
 
     @Test
-    public void whenPartition_callsGeneric() {
-        HazelcastInstance hz = createHazelcastInstance();
-        OperationService operationService = getOperationService(hz);
+    public void invokeOnPartition_outerGeneric_innerGeneric_forbidden() {
+        HazelcastInstance local = createHazelcastInstance();
+        OperationService operationService = getOperationService(local);
+
+        InnerOperation innerOperation = new InnerOperation(RESPONSE, GENERIC_OPERATION);
+        OuterOperation outerOperation = new OuterOperation(innerOperation, GENERIC_OPERATION);
+
+        expected.expect(Exception.class);
```

Comment:
Thx. I know this one - but it's not specific enough IMHO in many cases. 
You don't really know which line threw the exception. 
It's also impossible to match what the message was which is useful in some cases, where the same type of exception may be thrown twice from the same method, like it's matched here:
https://github.com/hazelcast/hazelcast/pull/6240/files#diff-efd38b559d7c6a3cc584ef18e630e672R79

There are cases where notation proposed by you is OK, but I don't see a reason why we should be dogmatic about it ;)